src/components/Dialogs/DistanceMeasuringDialog/DistanceMeasuringDialog.tsx

```diff
@@ -0,0 +1,256 @@
+import * as React from "react";
+import {observable, action, makeObservable} from "mobx";
+import {observer} from "mobx-react";
+import {ColorResult} from "react-color";
+import {FormGroup, IDialogProps, NonIdealState, H5} from "@blueprintjs/core";
+import * as AST from "ast_wrapper";
+import {AppStore, DialogStore, HelpType} from "stores";
+import {DistanceMeasuringStore, RegionCoordinate} from "stores/Frame";
+import {CustomIcon} from "icons/CustomIcons";
+import {SafeNumericInput, ColorPickerComponent, CoordinateComponent} from "components/Shared";
+import {DraggableDialogComponent} from "components/Dialogs";
+import {WCSPoint2D, Point2D} from "models";
+import {getPixelValueFromWCS, getFormattedWCSPoint, isWCSStringFormatValid, SWATCH_COLORS} from "utilities";
+import {ImageViewLayer} from "components";
+import "./DistanceMeasuringDialog.scss";
+
+const KEYCODE_ENTER = "Enter";
+
+@observer
+export class DistanceMeasuringDialog extends React.Component {
+    constructor(props: any) {
+        super(props);
+        makeObservable(this);
+    }
+
+    @observable WCSMode: boolean = true;
+
+    @action setWCSMode = (bool?: boolean) => {
```

Comment:
rename to `val` or `value`. `bool` is confusing as a variable name

Reply:
resolved